From: Tony Espy <espy@canonical.com>
Date: Sat, 5 Apr 2014 15:37:11 -0400
Subject: src: fix combined gprs-context property logic

This change modifies the gprs combined context ( Internet/MMS )
logic such that the MessageCenter and MessageProxy properties
are only added to a combined context if non-empty.  This differs
from the MMS logic where these properties will be added even if
the values from the context are zero-length strings.
---
 ofono/src/gprs.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/ofono/src/gprs.c b/ofono/src/gprs.c
index 8211dd5..d225d73 100644
--- a/ofono/src/gprs.c
+++ b/ofono/src/gprs.c
@@ -1094,7 +1094,8 @@ static void append_context_properties(struct pri_context *ctx,
 	ofono_dbus_dict_append(dict, "AuthenticationMethod", DBUS_TYPE_STRING,
 				&strvalue);
 
-	if (ctx->message_center != NULL) {
+	if (ctx->type == OFONO_GPRS_CONTEXT_TYPE_MMS ||
+		(ctx->message_center && strlen(ctx->message_center) > 0)) {
 		strvalue = ctx->message_proxy;
 		ofono_dbus_dict_append(dict, "MessageProxy",
 					DBUS_TYPE_STRING, &strvalue);
@@ -2340,7 +2341,8 @@ static void write_context_settings(struct ofono_gprs *gprs,
 	g_key_file_set_boolean(gprs->settings, context->key, "Preferred",
 				context->preferred);
 
-	if (context->message_center != NULL) {
+	if (context->type == OFONO_GPRS_CONTEXT_TYPE_MMS ||
+		(context->message_center && strlen(context->message_center) > 0)) {
 		g_key_file_set_string(gprs->settings, context->key,
 					"MessageProxy",
 					context->message_proxy);
